golang/cmd/kafka-to-postgresql-v2/kafka/kafka.go

```diff
@@ -10,43 +10,55 @@ import (
 	"math/rand"
 	"strconv"
 	"strings"
-	"sync"
+	"sync/atomic"
 )
 
+type IConnection interface {
+	GetMessages() <-chan *shared.KafkaMessage
+	MarkMessage(message *shared.KafkaMessage)
+	GetMarkedMessageCount() uint64
+}
+
 type Connection struct {
 	consumer *redpanda.Consumer
+	marked   atomic.Uint64
+}
+
+func (c *Connection) GetMarkedMessageCount() uint64 {
+	return c.marked.Load()
 }
 
 var conn *Connection
-var once sync.Once
 
-func GetOrInit() *Connection {
-	once.Do(func() {
-		zap.S().Debugf("kafka.GetOrInit().once")
-		KafkaBrokers, err := env.GetAsString("KAFKA_BROKERS", true, "http://united-manufacturing-hub-kafka.united-manufacturing-hub.svc.cluster.local:9092")
-		if err != nil {
-			zap.S().Fatalf("Failed to get KAFKA_BROKERS from env")
-		}
+func GetKafkaClient() IConnection {
+	return conn
+}
 
-		brokers := strings.Split(KafkaBrokers, ",")
-		instanceID := rand.Int63() //nolint:gosec
+func InitKafkaClient() {
+	zap.S().Debugf("kafka.GetKafkaClient()")
+	KafkaBrokers, err := env.GetAsString("KAFKA_BROKERS", true, "http://united-manufacturing-hub-kafka.united-manufacturing-hub.svc.cluster.local:9092")
+	if err != nil {
+		zap.S().Fatalf("Failed to get KAFKA_BROKERS from env")
+	}
 
-		consumer, err := redpanda.NewConsumer(brokers, []string{"^umh\\.v1.+$"}, "kafka-to-postgresql-v2", strconv.FormatInt(instanceID, 10))
-		if err != nil {
-			zap.S().Fatalf("Failed to create kafka client: %s", err)
-		}
-		conn = &Connection{
-			consumer: consumer,
-		}
-	})
-	return conn
+	brokers := strings.Split(KafkaBrokers, ",")
+	instanceID := rand.Int63() //nolint:gosec
+
+	consumer, err := redpanda.NewConsumer(brokers, []string{"^umh\\.v1.+$"}, "kafka-to-postgresql-v2", strconv.FormatInt(instanceID, 10))
+	if err != nil {
+		zap.S().Fatalf("Failed to create kafka client: %s", err)
+	}
+	conn = &Connection{
+		consumer: consumer,
+	}
 }
 
 func (c *Connection) GetMessages() <-chan *shared.KafkaMessage {
 	return c.consumer.GetMessages()
 }
 
 func (c *Connection) MarkMessage(message *shared.KafkaMessage) {
+	c.marked.Add(1)
```

Comment:
what is going on here? can we split up this PR into different bug fixes?

Reply:
yes